src/tracks/playabletrack/wavetrack/ui/WaveTrackAffordanceControls.cpp

```diff
@@ -221,7 +223,22 @@ std::vector<UIHandlePtr> WaveTrackAffordanceControls::HitTest(const TrackPanelMo
                 break;
             }
         }
-        mFocusClip = focusClip;
+        if(focusClip.use_count() > 0)
+            mFocusClip = focusClip;
+        else
+        {
+            //no clip under the cursor, show selection handle
+            const auto& settings = ProjectSettings::Get(*pProject);
+            const auto currentTool = settings.GetTool();
+            if (currentTool == ToolCodes::multiTool || currentTool == ToolCodes::selectTool)
+            {
+                results.push_back(
+                    SelectHandle::HitTest(
+                        mSelectHandle, state, pProject, std::static_pointer_cast<TrackView>(track->GetTrackView())
+                    )
+                );
+            }
```

Comment:
Likewise here, you might push SelectHandle always, but sometimes it is not the first hit.